tests/utilities/simnode.py

```diff
@@ -0,0 +1,94 @@
+import subprocess
+import argparse
+import yaml
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+schema = {
+    "type": "object",
+    "properties": {
+        "devices": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "nos": {
+                        "type": "string"
+                    },
+                    "version": {
+                        "type": "string"
+                    },
+                    "hostname": {
+                        "type": "string"
+                    },
+                    "ssh_port": {
+                        "type": "integer"
+                    },
+                    "rest_port": {
+                        "type": "integer"
+                    },
+                },
+                "required": [
+                    "nos", "version", "hostname"
+                ]
+            }
+        }
+    }
+}
+
+
+def run_simnodes(devices: list):
```

Comment:
Sorry for being pedantic, but can you please add docstrings?

Reply:
No problems, should be done now